subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishJavaIntegTest.groovy

```diff
@@ -18,13 +18,126 @@ package org.gradle.api.publish.maven
 
 import org.gradle.integtests.fixtures.publish.maven.AbstractMavenPublishIntegTest
 import org.gradle.test.fixtures.maven.MavenDependencyExclusion
+import org.gradle.test.fixtures.maven.MavenFileModule
+import org.gradle.test.fixtures.maven.MavenPublishedJavaModule
+import spock.lang.Unroll
 
 class MavenPublishJavaIntegTest extends AbstractMavenPublishIntegTest {
     def mavenModule = mavenRepo.module("org.gradle.test", "publishTest", "1.9")
+    def javaLibrary = new MavenPublishedJavaModule(mavenModule)
+
+    def useModuleMetadata = false
+
+    private void useModuleMetadata() {
+        executer.withArgument("-Dorg.gradle.internal.publishJavaModuleMetadata")
+        mavenModule.withModuleMetadata()
+        useModuleMetadata = true
+    }
+
+    def "can publish java-library with no dependencies"() {
+        createBuildScripts("""
+            publishing {
+                publications {
+                    maven(MavenPublication) {
+                        from components.java
+                    }
+                }
+            }
+""")
+
+        when:
+        run "publish"
+
+        then:
+        mavenModule.assertPublishedAsJavaModule()
+        mavenModule.parsedPom.scopes.isEmpty()
+
+        and:
+        resolveArtifacts(mavenModule) == ["publishTest-1.9.jar"]
+    }
+
+    def "can publish java-library with no dependencies using Gradle metadata"() {
+        useModuleMetadata()
+
+        createBuildScripts("""
+            publishing {
+                publications {
+                    maven(MavenPublication) {
+                        from components.java
+                    }
+                }
+            }
+""")
+
+        when:
+        run "publish"
+
+        then:
+        javaLibrary.assertPublished()
```

Comment:
:x: I had to read the test and the `MavenPublishedJavaModule` class back and forth multiple times to understand the interaction between the 2, and realize that in some tests you call `mavenModule.assertPublished()` and some others `javaLibrary.assertPublished()`. If possible I'd prefer a single Maven publish test fixture, that is configured depending on whether we use module metadata or not.

Reply:
I changed it so that `MavenPublishedJavaModule` now implements `MavenModule`, and just adds some convenience methods for verifying the published state of a java module.